hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -1031,6 +1031,19 @@ public static long getMultipartSizeProperty(Configuration conf,
     return partSize;
   }
 
+  public static boolean checkDiskBuffer(Configuration conf){
+    boolean isAllowedMultipart = conf.getBoolean(ALLOW_MULTIPART_UPLOADS,
+        IS_ALLOWED_MULTIPART_UPLOADS_DEFAULT);
+    if (isAllowedMultipart) {
```

Comment:
this is wrong here I guess. 
if isAllowedMultipart is enabled then FAST_UPLOAD_BUFFER must be disk else we throw an error right?

Reply:
If multipart is disabled and the FAST_UPLOAD_BUFFER is not disk then we throw an error.